src/components/TVChartContainer/SaveLoadAdapter.ts

```diff
@@ -0,0 +1,69 @@
+import { ChartData } from "charting_library";
+import { getTokenBySymbol } from "config/tokens";
+import { Token } from "domain/tokens";
+
+export class SaveLoadAdapter {
+  charts: ChartData[] | undefined;
+  setTvCharts: (a: ChartData[]) => void;
+  onSelectToken: (token: Token) => void;
+  chainId: number;
+
+  constructor(
+    chainId: number,
+    charts: ChartData[] | undefined,
+    setTvCharts: (a: ChartData[]) => void,
+    onSelectToken: (token: Token) => void
+  ) {
+    this.charts = charts;
+    this.setTvCharts = setTvCharts;
+    this.chainId = chainId;
+    this.onSelectToken = onSelectToken;
+  }
+
+  getAllCharts() {
+    return Promise.resolve(this.charts);
+  }
+
+  removeChart(id: string) {
+    if (!this.charts) return Promise.reject();
+    for (let i = 0; i < this.charts.length; ++i) {
+      if (this.charts[i].id === id) {
+        this.charts.splice(i, 1);
+        this.setTvCharts(this.charts);
+        return Promise.resolve();
+      }
+    }
+
+    return Promise.reject();
+  }
+
+  saveChart(chartData) {
+    if (!chartData.id) {
+      chartData.id = Math.random().toString();
+    } else {
+      this.removeChart(chartData.id);
```

Comment:
if `this.charts` is falsie then `removeChart` will be rejected. should `saveChart` be rejected in this case as well?

Reply:
and is there a reason why chart is removed here? few lines later it will be pushed to `this.charts`

Reply:
Regarding this I took their sample implementation and used it. I didn't want change because it can have an edge case issue or something.